genesyscloud/integration_action/resource_genesyscloud_integration_action_utils.go

```diff
@@ -0,0 +1,186 @@
+			}
+		}
+	}
+	return &platformclientv2.Requestconfig{}
+}
+
+// buildSdkActionConfigResponse takes the resource data and builds the SDK platformclientv2.Responseconfig from it
+func buildSdkActionConfigResponse(d *schema.ResourceData) *platformclientv2.Responseconfig {
+	if configResponse := d.Get("config_response"); configResponse != nil {
+		if configList := configResponse.([]interface{}); len(configList) > 0 {
+			configMap := configList[0].(map[string]interface{})
+
+			transMap := map[string]string{}
+			if mapVal, ok := configMap["translation_map"]; ok && mapVal != nil {
+				for key, val := range mapVal.(map[string]interface{}) {
+					transMap[key] = val.(string)
+				}
+			}
+			transMapDefaults := map[string]string{}
+			if mapVal, ok := configMap["translation_map_defaults"]; ok && mapVal != nil {
+				for key, val := range mapVal.(map[string]interface{}) {
+					transMapDefaults[key] = val.(string)
+				}
+			}
+			var successTemplate string
+			if tempVal, ok := configMap["success_template"]; ok {
+				successTemplate = tempVal.(string)
+			}
+
+			return &platformclientv2.Responseconfig{
+				TranslationMap:         &transMap,
+				TranslationMapDefaults: &transMapDefaults,
+				SuccessTemplate:        &successTemplate,
+			}
+		}
+	}
+	return &platformclientv2.Responseconfig{}
+}
+
+// flattenActionContract converts the custom ActionContract into a JSON-encoded string
+func flattenActionContract(schema interface{}) (string, diag.Diagnostics) {
+	if schema == nil {
+		return "", nil
+	}
+	schemaBytes, err := json.Marshal(schema)
+	if err != nil {
+		return "", diag.Errorf("Error marshalling action contract %v: %v", schema, err)
+	}
+	return string(schemaBytes), nil
+}
+
+// flattenActionConfigRequest converts the platformclientv2.Requestconfig into a map
+func flattenActionConfigRequest(sdkRequest platformclientv2.Requestconfig) []interface{} {
+	requestMap := make(map[string]interface{})
+	if sdkRequest.RequestUrlTemplate != nil {
+		requestMap["request_url_template"] = *sdkRequest.RequestUrlTemplate
+	}
+	if sdkRequest.RequestType != nil {
+		requestMap["request_type"] = *sdkRequest.RequestType
+	}
+	if sdkRequest.RequestTemplate != nil {
+		requestMap["request_template"] = *sdkRequest.RequestTemplate
+	}
+	if sdkRequest.Headers != nil {
+		requestMap["headers"] = *sdkRequest.Headers
+	}
+	return []interface{}{requestMap}
+}
+
+// flattenActionConfigResponse converts the the platformclientv2.Responseconfig into a map
+func flattenActionConfigResponse(sdkResponse platformclientv2.Responseconfig) []interface{} {
+	responseMap := make(map[string]interface{})
+	if sdkResponse.TranslationMap != nil {
+		responseMap["translation_map"] = *sdkResponse.TranslationMap
+	}
+	if sdkResponse.TranslationMapDefaults != nil {
+		responseMap["translation_map_defaults"] = *sdkResponse.TranslationMapDefaults
+	}
+	if sdkResponse.SuccessTemplate != nil {
+		responseMap["success_template"] = *sdkResponse.SuccessTemplate
+	}
```

Comment:
There is also a function in the resourcedata package for this logic: `SetMapValueIfNotNil`

Reply:
Used resourcedata util functions where applicable